tests/karma/beatmapset-search-filters.spec.ts

```diff
@@ -2,11 +2,21 @@
 // See the LICENCE file in the repository root for full licence text.
 
 import { BeatmapsetSearchFilters } from 'beatmapset-search-filters';
+import core from 'osu-core-singleton';
+import testCurrentUserJson from './test-current-user-json';
 
 describe('BeatmapsetSearchFilters', () => {
   let subject: BeatmapsetSearchFilters;
 
   describe('.query', () => {
+    beforeAll(() => {
+      core.setCurrentUser(testCurrentUserJson);
+    });
+
+    afterAll(() => {
+      core.setCurrentUser({ id: undefined });
+    });
```

Comment:
what's this for here?

Reply:
actual problem was if tests that simulated having a user ran first and didn't have `user_preferences` set left it undefined when logging out